src/envoy/http/mixer/report_data.h

```diff
@@ -72,10 +72,10 @@ class ReportData : public ::istio::control::http::ReportData,
         response_total_size_(info.bytesSent()),
         request_total_size_(request_total_size) {
     if (response_headers != nullptr) {
-      response_total_size_ += response_headers->byteSize();
+      response_total_size_ += response_headers->refreshByteSize();
```

Comment:
https://github.com/istio/proxy/pull/2447 uses this instead:
```
response_total_size_ += (response_headers->byteSize().has_value()
                             ? response_headers->byteSize().value()
                             : response_headers->byteSizeInternal());
```
not sure what's the reason for using `byteSizeInternal()`, to be honest.

cc @lambdai @asraa

Reply:
Ok. I think this is some const mess.

Reply:
Yeah, this is a WIP to clear this mess (see https://github.com/envoyproxy/envoy/issues/8567)
But generally the pattern would be to refresh the byte size before you report data / log data, and then use `byteSize()` inside these functions since they typically take const pointers / refs.

Reply:
yep. refreshByteSize is not allowed on const HeaderMap. How could this compiled?